Jenkinsfile.groovy

```diff
@@ -0,0 +1,143 @@
+ * GNU Affero General Public License along with this program.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+def dockerRegistry = "ghcr.io"
+def githubRepo = "icgc-argo/rdpc-ui"
+def commit = "UNKNOWN"
+def version = "UNKNOWN"
+
+pipeline {
+    kubernetes {
+        label 'platform-ui-executor'
+        yaml: """
+apiVersion: v1
+kind: Pod
+spec:
+  containers:
+  - name: node
+    image: node:16.13.0
+    tty: true
+  - name: docker
+    image: docker:18-git
+    tty: true
+    env:
+    - name: DOCKER_HOST
+      value: tcp://localhost:2375
+    - name: HOME
+      value: /home/jenkins/agent
+  - name: dind-daemon
+    image: docker:18.06-dind
+    securityContext:
+      privileged: true
+      runAsUser: 0
+    volumeMounts:
+    - name: docker-graph-storage
+      mountPath: /var/lib/docker
+  securityContext:
+    runAsUser: 1000
+  volumes:
+  - name: docker-graph-storage
+    emptyDir: {}
+"""
+    }
+
+    stages {
+        stage('Prepare') {
+            steps {
+                echo 'Prepare..'
+                script {
+                    commit = sh(returnStdout: true, script: 'git describe --always').trim()
+                    version = sh(returnStdout: true, script: 'cat ./package.json | grep version | cut -d \':\' -f2 | sed -e \'s/"//\' -e \'s/",//\'').trim()
+                }
+            }
+        }
+
+        stage('Test') {
+            steps {
+                echo 'Test...'
+                container('node') {
+                    sh "npm ci && npm t"
+
+                }
+            }
+        }
+
+        stage('Build container') {
+            steps {
+                container('docker') {
+                    // DNS error if --network is default
+                    sh "docker build --network=host -f Dockerfile . -t ${dockerRegistry}/${githubRepo}:${commit}"
+                }
+            }
+        }
+
+        stage('Deploy to argo-dev') {
+            when {
+                branch "develop"
+            }
+            steps {
+                container('docker') {
+                    withCredentials([usernamePassword(credentialsId: 'argoContainers', usernameVariable: 'USERNAME', passwordVariable: 'PASSWORD')]) {
```

Comment:
i think this injects credentials from our kubernetes env?

Reply:
Close! they're injected from the Jenkins environment.
These for instance are simple Github credentials for an account with permissions to the Argo org.